pyo3-derive-backend/src/pyclass.rs

```diff
@@ -92,18 +92,17 @@ impl PyClassArgs {
                 self.freelist = Some(syn::Expr::clone(right));
             }
             "name" => match &**right {
-                syn::Expr::Lit(
-                    lit
-                    @
-                    syn::ExprLit {
-                        lit: syn::Lit::Str(..),
-                        ..
-                    },
-                ) => {
-                    self.name = Some(lit.clone().into());
+                syn::Expr::Lit(syn::ExprLit {
+                    lit: syn::Lit::Str(lit),
+                    ..
+                }) => {
+                    self.name = match lit.parse() {
+                        Ok(name) => Some(name),
+                        Err(..) => expected!("type name (e.g., Name or \"Name\")"),
+                    }
                 }
                 syn::Expr::Path(exp) if exp.path.segments.len() == 1 => {
-                    self.name = Some(exp.clone().into());
+                    self.name = Some(exp.clone());
                 }
```

Comment:
I think this is the branch for `name` without quotes? I would actually be quite happy to deprecate this as I think `name="CustomName"` is clearer than `name=CustomName`.

If you agree, we could make this return an error message suggesting the user wrap the name in quotes. 

cc @kngwyu - do you agree with a breaking change here to tidy this up?

Reply:
I’d support that as well, also `ExprPath` is actually too restrictive as it doesn’t play well with substituted macro variables which are basically `None` delimited `ExprGroup`. That was the very reason for my initial PR.

Reply:
Yeah, I agree to remove the branch.